### PR TITLE
Fix DevSetupAgent logging and some other fixes.

### PR DESCRIPTION
**Pull request by sshilov7**
## Summary of the pull request
DevSetupAgent and DevSetupEngine were crashing on start after previous (#2474) change to logging setup.
This change
- Fixes DevSetupAgent and DevSetupEngine logging to output loggs into 
%localappdata%\Temp\DevSetupEngine\Logs\HyperV directory (like C:\Users\LocalAdmin\AppData\Local\Temp\DevSetupEngine\Logs\HyperV for DevSetupgent)
- Adds COM access permission to DevSetupAgent to allow calls from LocalSystem and Interactive Users only.
- Move Hyper-V DevSetupAgent PS script into a separate file (DevSetupAgent.ps1)
- Move HResultException.cs to HyperVExtension.HostGuestCommunication library to use on DevSetupAgent side
- Small fixes to error message in Host-Guest communication

## Validation steps performed
Tested DevSetupAgent setup, start and configuration workflow on a Hyper-V VM.

## PR checklist
- [ ] Closes #xxx
- [ ] Tests added/passed
- [ ] Documentation updated

### CODE REVIEW

File: HyperVExtension/src/HyperVExtension/Helpers/DevSetupAgentDeploymentHelper.cs [hunk 1]
```diff
@@ -124,203 +126,9 @@ private ushort GetVMArchitechture(PSObject session)
...
-                Remove-Item -Recurse -Force -Path $using:guestUnpackDirectory -ErrorAction SilentlyContinue 
-            }
-        }
-
-        Remove-PSSession $session
-}
-";
+    private static string LoadScript()
+    {
+        var path = Path.Combine(AppContext.BaseDirectory, @"HyperVExtension\Scripts\", "DevSetupAgent.ps1");
```
Comment (dhoehna) [inline HyperVExtension/src/HyperVExtension/Helpers/DevSetupAgentDeploymentHelper.cs:131]: NIT: Path.Join instead of '/'

**Review by dhoehna — APPROVED** [review]

**Comment by krschau** [comment]
I think we should keep all the Logs under the Logs folder, so instead of 
`%localappdata%\Temp\DevSetupEngine\Logs\HyperV`
it would be
`%localappdata%\TempState\Logs\DevSetupEngine\HyperV`

**Comment by sshilov7** [comment]
@krschau , I don't understand what you are asking for referring to [devhome/extensions/CoreWidgetProvider/Program.cs](https://github.com/microsoft/devhome/blob/13fe40ee5b81bd436f819381f80300b2704a5f4b/extensions/CoreWidgetProvider/Program.cs#L17). Please clarify.

As for the actual location: %localappdata%\TempState normally does not exist and is not documented. Currently, Dev Home logs go under %localappdata%\Packages\<PackageFamilyName>\TempState\Logs. Code changed in this PR is not part of Dev Home packaged app and doesn't have package identity and cannot write logs under that location. Furthermore, DevSetupAgent runs under LocalSystem account and cannot write logs under some user's account where Dev Home may exist.

File: HyperVExtension/src/DevSetupAgent/Program.cs [hunk 2]
```diff
@@ -1,41 +1,109 @@
...
+
+        uint absoluteSdSize = default;
+        uint daclSize = default;
+        uint saclSize = default;
+        uint ownerSize = default;
+        uint groupSize = default;
+
+        if (PInvoke.MakeAbsoluteSD(securityDescriptor, absolutSd, ref absoluteSdSize, null, ref daclSize, null, ref saclSize, ownerSid, ref ownerSize, groupSid, ref groupSize))
+        {
+            throw new HResultException(HRESULT.E_UNEXPECTED);
```
Comment (huzaifa-d) [inline HyperVExtension/src/DevSetupAgent/Program.cs:48]: Suggestion: Add a brief description of the error and where it failed.

**Review by huzaifa-d — APPROVED** [review]

**Comment by krschau** [comment]
> @krschau , I don't understand what you are asking for referring to [devhome/extensions/CoreWidgetProvider/Program.cs](https://github.com/microsoft/devhome/blob/13fe40ee5b81bd436f819381f80300b2704a5f4b/extensions/CoreWidgetProvider/Program.cs#L17). Please clarify.
> 
> As for the actual location: %localappdata%\TempState normally does not exist and is not documented. Currently, Dev Home logs go under %localappdata%\Packages<PackageFamilyName>\TempState\Logs. Code changed in this PR is not part of Dev Home packaged app and doesn't have package identity and cannot write logs under that location. Furthermore, DevSetupAgent runs under LocalSystem account and cannot write logs under some user's account where Dev Home may exist.

Sorry, I assumed this was part of the extension and could write to the Dev Home logs.

**Pull request merged**